weave/trace/weave_init.py

```diff
@@ -207,8 +208,15 @@ def init_weave_disabled() -> InitializedClient:
 def init_weave_get_server(
     api_key: str | None = None,
     should_batch: bool = True,
-) -> remote_http_trace_server.RemoteHTTPTraceServer:
-    res = remote_http_trace_server.RemoteHTTPTraceServer.from_env(should_batch)
+) -> (
+    remote_http_trace_server.RemoteHTTPTraceServer
+    | stainless_trace_server.RemoteHTTPTraceServer
```

Comment:
Shouldn't we abstract the interface as an abstract base class? I guess essentially `TraceServerInterface`?